src/procman-app.cpp

```diff
@@ -308,8 +308,17 @@ procman_data_new (GSettings *settings)
     g_free (color);
 
     /* Sanity checks */
-    swidth = WidthOfScreen (gdk_x11_screen_get_xscreen (gdk_screen_get_default ()));
-    sheight = HeightOfScreen (gdk_x11_screen_get_xscreen (gdk_screen_get_default ()));
+#ifdef GDK_WINDOWING_X11
+    if (GDK_IS_X11_DISPLAY (gdk_display_get_default ())) {
+        swidth = WidthOfScreen (gdk_x11_screen_get_xscreen (gdk_screen_get_default ()));
+        sheight = HeightOfScreen (gdk_x11_screen_get_xscreen (gdk_screen_get_default ()));
+    }
+#else
+    /* For now, hardcode the initial width/height on Wayland */
+    swidth = 50;
+    sheight = 50;
+#endif
```

Comment:
I think we can use gdkscreen. An example where https://github.com/mate-desktop/caja-actions/commit/5055b5984372f09608d568680f2382f447a9b17e was replaced with Width/HeightOfScreen, (unluckily)
https://github.com/mate-desktop/caja-actions/commit/39e5471548f48c90eb2bfcd39024448e44424359
or with GdKPoint (maybe, i am not sure).
https://github.com/mate-desktop/mate-panel/commit/96ceed2717780188bc0eb84565d91617783d7192